base/stat.jl

```diff
@@ -326,6 +326,18 @@ otherwise returns `false`.
 This is the generalization of [`isfile`](@ref), [`isdir`](@ref) etc.
 """
 ispath(st::StatStruct) = filemode(st) & 0xf000 != 0x0000
+function ispath(path::String)
+    if contains(path, '\0')
+        throw(ArgumentError("embedded NULs are not allowed in C strings: \"$(repr(path))\""))
+    end
+    # We use `access()` and `F_OK` to determine if a given path exists.
+    # `F_OK` comes from `unistd.h`.
+    F_OK = 0x00
+    res = ccall(:jl_fs_access, Cint, (Ptr{UInt8}, Cint), path, F_OK)
+    @assert res in (0, Base.UV_ENOENT)
```

Comment:
I can't get the tests to pass without this assert here. My theory is that there is one function in `base/file.jl` that relies on this throwing.

Reply:
Nevermind, must have been a fluke from `test-revise`, because after a rebuild its working now.

Reply:
The problem was that I did not handle access denied errors and such.